src/ReactDOMRe.re

```diff
@@ -1124,16 +1160,19 @@ type domProps = {
 
 [@bs.splice] [@bs.module "react"]
 external createDOMElementVariadic:
-  (string, ~props: domProps=?, array(React.element)) => React.element =
+  (string, ~props: domProps(_)=?, array(React.element)) => React.element =
   "createElement";
 
 /* This list isn't exhaustive. We'll add more as we go. */
 /*
  * Watch out! There are two props types and the only difference is the type of ref.
  * Please keep in sync.
  */
+
+/** [props('input)] is the type of the React element attributes object.
+    ['input] is the type of values accepted by [input] elements. */
 [@bs.deriving abstract]
-type props = {
+type props('input) = {
```

Comment:
@rickyvetter mentioned yesterday that `props` is for JSX2, maybe we should consider not adding the 'type-safe' input support to JSX2?

Reply:
So `props` can be left alone as it is likely to be removed soon, right?

Reply:
Yes. `props` will be deprecated in 0.9 and removed in 0.10/1.0.